dev/com.ibm.ws.ejbcontainer.mdb/src/com/ibm/ws/ejbcontainer/mdb/internal/MessageEndpointFactoryImpl.java

```diff
@@ -108,7 +116,29 @@ public void initialize(EJSContainer ejsContainer,
         // binding file, or defaulted to "[app/]module/bean" conforming to the
         // java:global rules for binding an EJB.
         activationSvcId = bmd.ivActivationSpecJndiName;
-        name = "messageEndpoint-" + activationSvcId;
+
+        //Register this instance as a PauseableComponent
+
+        name = bmd.enterpriseBeanName;
+        if (name != null && bmd.getModuleMetaData().getName() != null) {
```

Comment:
I like that this is using the full Java EE qualified name; however, it should be consistent with java:global naming conventions and the default JNDI activation spec names.  Specifically, it should use the 'name' of the module as indicated in XML (or default to name of JAR/WAR without the extension) and should NOT include an application name if the WAR/JAR is installed standalone (i.e. not in an EAR).  Specifically, it should use mmd.ivLogicalName for the module name and amd.getLogicalName() for the application.  See EJBMDOrchestratorImpl.getDefaultActivationSpecJndiName for an example.  getModuleMetaData().getName() will return the file name of the .jar/.war including extension, whereas ivLogicalName will be the Java EE spec defined module name; similarly, getApplicationMetaData().getName will return the name of the EAR, or the 'internal' made up name of an EAR when an EAR doesn't actually exist; getApplicationMetaData().getLogicalName() will return null if the JAR/WAR isn't in an EAR, or the Java EE spec defined name if it is in an EAR.